deepinv/optim/potential.py

```diff
@@ -0,0 +1,162 @@
+import torch
+import torch.nn as nn
+import deepinv
+from deepinv.optim.utils import gradient_descent
+
+
+class Potential(nn.Module):
+    r"""
+    Base class for a potential :math:`h : \xset \to \mathbb{R}` to be used in an optimization problem.
+
+    Comes with methods to compute the potential gradient, its proximity operator, its convex conjugate (and associated gradient and prox).
+
+    :param callable fn: Potential function :math:`h(x)` to be used in the optimization problem.
+    """
+
+    def __init__(self, fn=None):
```

Comment:
```suggestion
    def __init__(self, fn):
```

Shouldn't fn be a mandatory argument?

Reply:
No because one could overwrite directly the forward instead of defining it with fn